Tests/SentryTests/Helper/SentryFileManagerTests.swift

```diff
@@ -525,6 +526,74 @@ class SentryFileManagerTests: XCTestCase {
         XCTAssertNil(actualSession)
     }
     
+    func testStoreAbnormalSession() throws {
+        // Arrange
+        let session = SentrySession(releaseName: "1.0.0", distinctId: "some-id")
+        session.abnormalMechanism = "anr_foreground"
+        
+        // Act
+        sut.storeAbnormalSession(session)
+        
+        // Assert
+        let actualSession = try XCTUnwrap(sut.readAbnormalSession())
+        
+        // Only assert a few properties. SentrySessionTests tests the serialization
+        XCTAssertEqual(session.sessionId, actualSession.sessionId)
+        XCTAssertEqual(session.distinctId, actualSession.distinctId)
+        XCTAssertEqual(session.releaseName, actualSession.releaseName)
+        XCTAssertEqual(session.abnormalMechanism, actualSession.abnormalMechanism)
+    }
+    
+    func testDeleteAbnormalSession() throws {
+        // Arrange
+        let session = SentrySession(releaseName: "1.0.0", distinctId: "some-id")
+        session.abnormalMechanism = "anr_foreground"
+        sut.storeAbnormalSession(session)
+        
+        // Act
+        sut.deleteAbnormalSession()
+        
+        // Assert
+        XCTAssertNil(sut.readAbnormalSession())
+    }
+    
+    func testDeleteAbnormalSession_WhenNoAbnormalSessionStored_DoesNotCrash() throws {
+        sut.deleteAbnormalSession()
+    }
+    
+    func testReadAbnormalSession_NoSessionStored () throws {
+        XCTAssertNil(sut.readAbnormalSession())
+    }
+    
+    func testAbnormalSessionAsync_DoesNotCrash() {
+        // Arrange
+        let iterations = 1_000
+        let expectation = expectation(description: "complete all abnormal session interactions")
+        expectation.expectedFulfillmentCount = iterations * 3
+        let dispatchQueue = DispatchQueue(label: "testAbnormalSessionAsync_DoesNotCrash", qos: .userInitiated, attributes: [.concurrent])
+        
+        // Act
+        for _ in 0..<iterations {
```

Comment:
`l`: it might be worth to bring in some variance by adding sleep statements (but not with random durations, as test input should not change in between invocations)

Reply:
This test aims to identify potential locking issues that could lead to a crash. While adding sleeps may make the test appear more realistic, it significantly reduces its effectiveness by decreasing the likelihood of read, write, and delete operations overlapping across different threads, thereby making it less likely to uncover concurrency issues.